### PR TITLE
Modify origin address for avatax calls

### PR DESCRIPTION
**Pull request by rajrishav0711**
[Jira](https://dekeo.atlassian.net/browse/ALPHA-666)

We are updating `shipFrom` field we send to avalara, as per the conditions mentioned in the story.

### CODE REVIEW

File: app/models/solidus_avatax_certified/line.rb [hunk 1]
```diff
@@ -122,24 +122,14 @@ def return_item_line(line_item, quantity, amount)
...
         }
       }.merge(base_line_hash)
 
       return_item_line.merge(tax_override_hash(amount))
     end
 
-    def get_stock_location(li)
-      inventory_units = li.inventory_units
-
-      return default_ship_from if inventory_units.blank?
```
Comment (ashishtajane) [inline app/models/solidus_avatax_certified/line.rb:136]: We need to get rid of `default_ship_from` method. And use company address in `refund_line` instead.
Let's confirm that from Ben and Don but that is how it should be I think.

File: app/models/solidus_avatax_certified/line.rb [hunk 2]
```diff
@@ -194,5 +184,45 @@ def shipment_cost(shipment)
...
     end
+
+    def canada_ship_from
+      {
+        line1: '420 Green St',
+        line2: 'Unit 206',
+        city: 'Whitby',
+        region: 'ON',
+        country: 'CA',
+        postalCode: '19801'
```
Comment (ashishtajane) [inline app/models/solidus_avatax_certified/line.rb]: Copy-paste error!

File: app/models/solidus_avatax_certified/line.rb [hunk 3]
```diff
@@ -194,5 +184,45 @@ def shipment_cost(shipment)
...
+                 elsif shipment.present?
+                   shipment.stock_location
+                 end
+     return usa_ship_from unless stock_loc || stock_loc.state.abbr.in?(['MO','PA'])
+
+     case type
+     when 'item_line', 'return_line'
+       stock_loc.to_avatax_hash
+     when 'shipment_line'
+       shipment.stock_location.to_avatax_hash
```
Comment (ashishtajane) [inline app/models/solidus_avatax_certified/line.rb]: Is this `case...when..` really needed?
Won't this be same as `stock_loc.to_avatax_hash`?

Also - can we please use full name for variable please

File: app/models/solidus_avatax_certified/line.rb [hunk 4]
```diff
@@ -38,7 +38,7 @@ def item_line(line_item)
         discounted: true,
         taxIncluded: false,
         addresses: {
-          shipFrom: get_stock_location(line_item),
+          shipFrom: ship_from_address('line_item', line_item),
```
Comment (ashishtajane) [inline app/models/solidus_avatax_certified/line.rb]: first argument should be `item_line`?

**Review by ashishtajane — APPROVED** [review]

**Comment by rajatbansal93** [comment]
Can we please add some more description here also

**Pull request merged**